core/src/main/java/io/micronaut/core/annotation/AnnotationExpressionReference.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.annotation;
+
+/**
+ * Wrapper for annotation value, containing evaluated expressions and
+ * class name for generated expression class. This class is only used
+ * at compilation time as part of compile-time annotation metadata.
+ *
+ * @author Sergey Gavrilov
+ * @since 4.0.0
+ */
+@Internal
+public final class AnnotationExpressionReference {
```

Comment:
can be a record

Reply:
if it is only used at compilation time can it be moved to the processor module?

Reply:
Turned this one intro a record. It can't be moved to processor module as in current implementation `MutableAnnotationMetadata`  check whether it has evaluated expressions by the presence of this class instances in its values